### ISSUE TITLE
Foreign Key constraint violation

### ISSUE THREAD

**Issue by sriechersrc** [payloadcms/payload#6845]
### Payload Version

3.0.0-beta.46

### Node Version

v20.10.0

### Next.js Version

15.0.0-rc.0

### Describe the Bug

I have been using the postgres adapter for a while now and have been running into a lot of foreign key constraint errors. Sadly I can't really tell when those errors are accuring, but they are quit frequent. 

The last one I caught was this one: `error: insert or update on table "events_locales" violates foreign key constraint "events_locales_parent_id_fk"` after adding a relationship field to my events collection.

```typescript
 ⨯ node_modules/.pnpm/pg-pool@3.6.2_pg@8.11.3/node_modules/pg-pool/index.js (45:1) @ eval
 ⨯ error: insert or update on table "events_locales" violates foreign key constraint "events_locales_parent_id_fk"
    at eval (webpack-internal:///(rsc)/./node_modules/.pnpm/pg-pool@3.6.2_pg@8.11.3/node_modules/pg-pool/index.js:45:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async DrizzleORMPgClient.query (/Users/stevenriechersrc/Desktop/PROJECTS/my-project/node_modules/.pnpm/drizzle-kit@0.20.14-1f2c838/node_modules/drizzle-kit/payload.js:34498:21)
    at async apply (/Users/stevenriechersrc/Desktop/PROJECTS/my-project/node_modules/.pnpm/drizzle-kit@0.20.14-1f2c838/node_modules/drizzle-kit/payload.js:36648:9)
    at async pushDevSchema (webpack-internal:///(rsc)/./node_modules/.pnpm/@payloadcms+db-postgres@3.0.0-beta.46_payload@3.0.0-beta.46_react@19.0.0-rc-f994737d14-202405_fjwknzmbrzvje7mytctp7ne6gi/node_modules/@payloadcms/db-postgres/dist/utilities/pushDevSchema.js:47:5)
    at async Object.connect (webpack-internal:///(rsc)/./node_modules/.pnpm/@payloadcms+db-postgres@3.0.0-beta.46_payload@3.0.0-beta.46_react@19.0.0-rc-f994737d14-202405_fjwknzmbrzvje7mytctp7ne6gi/node_modules/@payloadcms/db-postgres/dist/connect.js:87:9)
    at async BasePayload.init (webpack-internal:///(rsc)/./node_modules/.pnpm/payload@3.0.0-beta.46_@swc+core@1.5.28_@swc+types@0.1.8_graphql@16.8.1_typescript@5.4.5/node_modules/payload/dist/index.js:253:13)
    at async getPayload (webpack-internal:///(rsc)/./node_modules/.pnpm/payload@3.0.0-beta.46_@swc+core@1.5.28_@swc+types@0.1.8_graphql@16.8.1_typescript@5.4.5/node_modules/payload/dist/index.js:333:26)
    at async Object.generateStaticParams (webpack-internal:///(rsc)/./src/app/(frontend)/[...slug]/page.tsx:167:21)
    at async buildParams (/Users/stevenriechersrc/Desktop/PROJECTS/my-project/node_modules/.pnpm/next@15.0.0-rc.0_react-dom@19.0.0-rc-f994737d14-20240522_react@19.0.0-rc-f994737d14-20240522/node_modules/next/dist/build/utils.js:1037:40) {
  type: 'DatabaseError',
  length: 524,
  severity: 'ERROR',
  code: '23503',
  detail: 'Key (_parent_id)=(1) is not present in table "events".',
  where: 'SQL statement "ALTER TABLE "events_locales" ADD CONSTRAINT "events_locales_parent_id_fk" FOREIGN KEY ("_parent_id") REFERENCES "events"("id") ON DELETE cascade ON UPDATE no action"\n' +
    'PL/pgSQL function inline_code_block line 2 at SQL statement',
  schema: 'public',
  table: 'events_locales',
  constraint: 'events_locales_parent_id_fk',
  file: 'ri_triggers.c',
  line: '2608',
  routine: 'ri_ReportViolation',
  page: '/en'
}
  43 |     // replace the stack trace that leads to `TCP.onStreamRead` with one that leads back to the
  44 |     // application that created the query
> 45 |     Error.captureStackTrace(err)
     | ^
  46 |     throw err
  47 |   })
  48 |   return { callback: cb, result: result }
```

Also I cannot fix these errors which leads to me deleting the whole database and reinitalizing it. This might not be a big problem in dev, but in production this would be pretty bad

### Reproduction Steps

Can't really tell how to reproduce these errors :(

### Adapters and Plugins

db-postgres

**Comment by marvinengelmann** [comment]
Any updates on this? I also run into this errors very frequently. Every time it occurs I have to ```run dev``` again, otherwise, localhost will load infinitely. The issue occurs much more frequently when I open both /admin and the frontend at the same time. However, everything still works as expected; it just slows down the development process.

**Comment by starlight-danju** [comment]
same issue here🤣

**Comment by pdcolandrea** [comment]
I've noticed similar issues when doing bulk insertions on my side. Occasionally, restarting the development server, as another commenter suggested, solves the problem. Other times, if I leave it for a few hours, it starts working again. Wiping the database always fixes it, though. Was originally on beta-46; moved up to beta-65 and the issue still occurs

**Comment by alsherif-khalaf** [comment]
![image](https://github.com/user-attachments/assets/85198fc7-9b9d-419a-86c4-21e8ac7b9ae0)
same is here when using localize : true ,

**Comment by alsherif-khalaf** [comment]
```bash
Error Type: DatabaseError
Severity: ERROR
Code: 23503
Table: _pages_v_blocks_header1_buttons
Constraint: _pages_v_blocks_header1_buttons_parent_id_fk
Detail: Key (_parent_id)=(20) is not present in table "_pages_v_blocks_header1".
Where: SQL statement "ALTER TABLE "_pages_v_blocks_header1_buttons" ADD CONSTRAINT "_pages_v_blocks_header1_buttons_parent_id_fk" FOREIGN KEY ("_parent_id") REFERENCES "_pages_v_blocks_header1"("id") ON DELETE cascade ON UPDATE no action"
```

**Comment by mikegreiling** [comment]
Just adding my voice to the chorus here... This is a showstopping bug. I was keen to try out `create-payload-app` on the v3 beta for the first time as my team is evaluating Payload for our homepage redesign and this bug came up within the first minute as I was simply poking around with a freshly-seeded homepage.

This is quite bad. Once encountered, I cannot do anything. Visiting the admin page yields the error. Visiting the published home page shows the error. The only way to resolve this is to wipe the db manually. Really not good.

I realize this is a beta, so I'm tempering my criticism here, but this is a really bad first impression.

**Comment by mikegreiling** [comment]
^ this was on the latest `3.0.0-beta.71` by the way

**Comment by paulpopus** [comment]
Hey all, just wanna say thank you for your patience on this. We are going to get postgres stable for v3, and @DanRibbens has been rewriting parts of the postgres adapter ahead of releasing the sqlite one. That PR will be merged in very soon, and we will have a concerted effort to fix these postgres issues.

Rest assured that they're not being ignored 🙏

**Comment by DanRibbens** [comment]
Fixed in https://github.com/payloadcms/payload/pull/7453

**Comment by HarleySalas** [comment]
> Fixed in #7453

Thank you!

I have been having this issue for quite some time and never submitted an issue, because I didn't have time to investigate creating a consistent reproduction. It's been a major pain for so long.

Your work is *greatly* appreciated!

**Comment by przyb** [comment]
@DanRibbens same happens with `payload v2.25.0` with `@payloadcms/db-postgres 0.8.5`. Can we expect the same patch for v2?

**Comment by DanRibbens** [comment]
@przyb, I think things have settled out enough with the recent version bumps to drizzle-kit + drizzle-orm in beta. We can make another update to postgres for 2.x but it won't have everything that beta currently does since the branches have diverged too significantly with breaking changes.

Here is the PR I opened to evaluate what we can/cannot migrate back to 2.x: https://github.com/payloadcms/payload/pull/7460

**Comment by didiraja** [comment]
@DanRibbens Really not happy to be migrating because 2.x became buggy, invested a lot of time building the project in this version but response from last weeks to all problems is "update to beta version". Not exactly confident to solve bugs using a beta version.

**Comment by DanRibbens** [comment]
Alright @przyb and @didiraja,

I feel this. I'll work on it again.

**Comment by r1tsuu** [comment]
We ported all postgres fixes to 2.x from 3.0 with this [PR](https://github.com/payloadcms/payload/pull/8468), including drizzle version. It should be solved now with 2.30.2 https://github.com/payloadcms/payload/releases/tag/v2.30.2
